### PR TITLE
DL: Add Postgres support

### PR DESCRIPTION
**Pull request by njayaram2**
JIRA: MADLIB-1311

This PR does the following:
1) decouples GPDB specific code to enable Postgres support too
for Deep Learning modules.
2) Jenkins docker image now pulls down Postgres 10, and also has tensorflow
and Keras installed in it. The default cmake is good to build MADlib, so
we don't have to build it from source like how we did previously

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/832/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/833/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/839/

File: src/ports/postgres/modules/deep_learning/madlib_keras.py_in [hunk 1]
```diff
@@ -102,39 +103,48 @@ def fit(schema_madlib, source_table, model, dependent_varname,
...
 
-    # Compute total buffers on each segment
-    total_buffers_per_seg = plpy.execute(
-        """ SELECT gp_segment_id, count(*) AS total_buffers_per_seg
-            FROM {0}
-            GROUP BY gp_segment_id
-        """.format(source_table))
-    seg_nums = [int(each_buffer["gp_segment_id"])
-        for each_buffer in total_buffers_per_seg]
+    optimizers = get_optimizers()
```
[the remark this answers is not in the document]
Comment (njayaram2) [reply src/ports/postgres/modules/deep_learning/madlib_keras.py_in]: Hmm, good point, not sure why that is here. Will remove it.

**Pull request closed without merging**